### PR TITLE
Build the extract function using merl

### PR DESCRIPTION
**Pull request by andytill**
In the ddl compiler, build the extract function using merl instead of hand rolling AST.  There is no change in the output, only in the way it is constructed.

### CODE REVIEW

File: src/riak_ql_ddl_compiler.erl [hunk 1]
```diff
@@ -270,32 +270,21 @@ canonical_arg_fmt_v1(?SQL_PARAM{name = [Nm]}) -> binary_to_list(Nm).
 
 %% the funny shape of this function is because it used to handle maps
```
Comment (gordonguthrie) [inline src/riak_ql_ddl_compiler.erl]: I presume this comment should be deleted?

Comment (andytill) [reply src/riak_ql_ddl_compiler.erl]: Yes! Deleted in 000548c2fcaa8f2e048f7851375d5dad9024f8e5

**Comment by gordonguthrie** [comment]
+1 000548c

**Comment by hmmr** [comment]
+1

**Comment by hazen** [comment]
:+1:

**Pull request merged**